assets/js/modules/adsense/common/ad-blocker-warning.test.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * AdSense AdBlocker Warning component tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import AdBlockerWarning from './ad-blocker-warning';
+import { render } from '../../../../../tests/js/test-utils';
+import { STORE_NAME } from '../datastore/constants';
+
+const setupAdBlockerRegistry = async ( registry ) => {
+	await registry.dispatch( STORE_NAME ).receiveIsAdBlockerActive( true );
+};
+
+const setupNoAdBlockerRegistry = async ( registry ) => {
+	await registry.dispatch( STORE_NAME ).receiveIsAdBlockerActive( false );
+};
+
+describe( 'AdBlockerWarning', () => {
+	afterAll( () => jest.restoreAllMocks() );
+
+	it( 'should render the warning when an AdBlocker is active', async () => {
+		const { container } = render( <AdBlockerWarning />, { setupRegistry: setupAdBlockerRegistry } );
+
+		expect( container.querySelector( '.googlesitekit-settings-module-warning' ) ).not.toEqual( null );
```

Comment:
Testing to only not equaling `null` is a bit dangerous, since `undefined` or `''` would pass this test. It's not exactly doing what the test says; we should use `getByText` here to find the actual warning text and ensure it's rendered.

It's best not to us `querySelector` tests in general, as they're testing for what the machine sees, rather than the user, so it's a good sign that the test could be improved. 🙂